internal/controllers/rpc.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/NodeFactoryIo/vedran/internal/rpc"
+	log "github.com/sirupsen/logrus"
+)
+
+func (c ApiController) RPCHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	defer r.Body.Close()
+	reqBody, _ := ioutil.ReadAll(r.Body)
+	isBatch := rpc.IsBatch(reqBody)
+	var reqRPCBody rpc.RPCRequest
+	var reqRPCBodies []rpc.RPCRequest
+	var err error
+	if isBatch {
+		err = json.Unmarshal(reqBody, &reqRPCBodies)
+	} else {
+		err = json.Unmarshal(reqBody, &reqRPCBody)
+	}
+
+	if err != nil {
+		log.Errorf("Request failed because of: %v", err)
+		_ = json.NewEncoder(w).Encode(
+			rpc.CreateRPCError(isBatch, reqRPCBody, reqRPCBodies, rpc.ParseError, "Parse error"))
+		return
+	}
+
+	if !isBatch && reqRPCBody == (rpc.RPCRequest{}) {
+		log.Errorf("Invalid request")
```

Comment:
I think you should check for single and batch or just let node to propagate error